packages/chart-advisor/src/advisor/index.ts

```diff
@@ -114,7 +114,7 @@ export class Advisor {
    * chart advising from input data
    * @param params paramters for advising
    */
-  advise(params: AdviseParams): Advice[] {
+  advise(params: AdviseParams): Advice[] | AdvicesWithLog {
```

Comment:
Can `log` as an optional parameter, the type is represented by one?
Or add a `getExplain` method to the class?

Reply:
1. we should change this return type in one since v3.0.0 but not now. cause by doing that, we are making a Breaking Change to Advisor.advise
2. if we offer an independent method named `getExpain` or `getLog`, users have to call `advise` and `getLog`, it means all the computation would be done twice....
3. But there IS a better design: we could provide a method named `adviseWithLog`, so we would have two advising methods `Advisor.advise` and `Advisor.adviseWithLog`. If the user does want to get the log, just call the latter...what do you think?

Reply:
check it again~